ocaml/typing/env.ml

```diff
@@ -1183,9 +1214,14 @@ and expand_module_path lax env path =
     {mdl_type=MtyL_alias path1} ->
       let path' = normalize_module_path lax env path1 in
       if lax || !Clflags.transparent_modules then path' else
-      let id = Path.head path in
-      if Ident.is_global_or_predef id && not (Ident.same id (Path.head path'))
-      then add_required_global id;
+      let head = address_head (find_value_address path env) in
+      let head' = address_head (find_value_address path' env) in
+      begin match head with
+      | AHunit cu when not (equal_address_head head head') ->
+        add_required_global cu
+      | AHunit _ | AHlocal _ ->
```

Comment:
I think that this is adding the wrong thing to the required global. The point of this code is to make sure that a reference to `Foo.Bar` will force `Foo` to get linked -- even if `Bar` is a transparent alias to a module in another file. Whereas this new version asks for the address of `Foo.Bar` which will not be `Foo` if it's a transparent alias.

Reply:
Now that `add_required_global` takes a `Path.t`, I've put this back to essentially its original form.